### PR TITLE
Krill Helm Chart and CI Docker Builds

### PR DESCRIPTION
**Pull request by davidholsgrove**
Check in Cargo.lock to have more predictable builds

### CODE REVIEW

File: charts/krill/templates/tests/test-connection.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (zenc234) [inline charts/krill/templates/tests/test-connection.yaml:1]: Is the purpose of this pod to test whether the deployment is reachable through the service port?

Comment (davidholsgrove) [reply charts/krill/templates/tests/test-connection.yaml:1]: Yeah exactly - https://helm.sh/docs/topics/chart_tests/
So you can run `helm test krill` to deploy a test pod, which verifies the svc and deployment have successfully started the pod and it responds. Its part of the default Helm chart set

**Comment by zenc234** [comment]
Changes LGTM.

File: charts/krill/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          volumeMounts:
+            - name: krill-conf
+              mountPath: {{ .Values.krillConf.data_dir }}/krill.conf
+              subPath: krill.conf
+            - name: data
+              mountPath: {{ .Values.krillConf.data_dir }}
+              subPath: {{ .Values.persistence.subPath | default "" | quote }}
+            {{- if and .Values.persistence.enabled .Values.persistence.rrdp.enabled }}
+            - name: rrdp-data
+              mountPath: {{ .Values.krillConf.data_dir }}/rrdp
```
Comment (zenc234) [inline charts/krill/templates/deployment.yaml]: In the registry testbed, the krill instance publishes rrdp data under `./data/repo/rrdp`. Over here, the `rrdp-data` pv is mounted at `./data` instead.

**Review by zenc234 — APPROVED** [review]
Changes look good to me. Helm chart tested by deploying in the test cluster.

**Pull request merged**